microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -773,11 +799,37 @@ def verify_bash_history_is_empty(self, node: Node) -> None:
                 "/root/.bash_history is not empty, this image is not prepared well."
             ).is_empty()
 
+    @TestCaseMetadata(
+        description="""
+        This test will check for kernel error, failure, alert messages from demsg.
+        Kenrel only, errors and alerts only. Should never fail.
+
+        Steps:
+        1. Get failure, error, alert messages from dmesg.
+        2. If any unexpected failure, error, warning messages excluding ignorable ones
+         existing, fail the case.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_platform_type=[AZURE, READY]),
+    )
+    def verify_boot_error_fail_warnings_dmesg(self, node: Node) -> None:
+        dmesg = node.tools[Dmesg]
+        log_output = dmesg.get_output(
+            force_run=True, log_level=["warn", "err", "emerg", "alert", "crit"]
+        )
+        found_results = self._process_log_lines(log_output)
```

Comment:
Does it need to filter in the warn level messages? I remembered we hope the kernel message is clean to avoid confuse users. @LiliDeng

Reply:
I mean, when we have the warn and above levels, the listed string may not be necessary. So if there is any warn and above message, the test case should fail.